### PR TITLE
Return optional string in default_wallet_address

### LINKED ISSUE #2663: wallet_default_address RPC call should return an optional value
**Issue summary**

It's not always the case that we have a default wallet address when running the forest daemon. As such the `wallet_default_address` RPC command should return an `Option<String>`

Context: https://github.com/ChainSafe/forest/blob/a01166f367f38ca9da220f2497dddcfbc8b6d158/node/rpc/src/wallet_api.rs#L47

**Other information and links**

### PR DESCRIPTION
**Pull request by creativcoder**
## Summary of changes

Changes introduced in this pull request:

- Returns Option<String> for wallet_default_address RPC fn.

## Reference issue to close (if applicable)

Closes #2663 

## Change checklist

- [X] I have performed a self-review of my own code,
- [X] I have made corresponding changes to the documentation,
- [X] I have added tests that prove my fix is effective or that my feature works
      (if possible),
- [X] I have made sure the
      [CHANGELOG](https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md) is
      up-to-date. All user-facing changes should be reflected in this document.

### CODE REVIEW

File: forest/cli/src/cli/wallet_cmd.rs [hunk 1]
```diff
@@ -218,7 +219,10 @@ impl WalletCommands {
 
                 for address in response {
                     let addr = address.0.to_string();
-                    let default_address_mark = if addr == default { "X" } else { "" };
+                    let default_address_mark = default
+                        .as_ref()
+                        .map(|s| if s == &addr { "X" } else { "" })
+                        .unwrap_or("");
```
Comment (lemmih) [inline forest/cli/src/cli/wallet_cmd.rs:225]: Here we can do `default == Some(addr)`, right?

File: node/rpc/src/wallet_api.rs [hunk 2]
```diff
@@ -54,7 +54,7 @@ where
     let keystore = data.keystore.read().await;
 
     let addr = forest_key_management::get_default(&keystore)?;
-    Ok(addr.to_string())
+    Ok(Some(addr.to_string()))
```
Comment (lemmih) [inline node/rpc/src/wallet_api.rs:57]: How will this ever return `None?`

**Review by lemmih — CHANGES_REQUESTED** [review]

File: forest/cli/src/cli/wallet_cmd.rs (hunk 1 above)
Comment (creativcoder) [reply forest/cli/src/cli/wallet_cmd.rs:225]: i had that initially, but fmt makes it a bit long to parse, so used this. want me to revert to `if else`?

File: node/rpc/src/wallet_api.rs (hunk 2 above)
Comment (creativcoder) [reply node/rpc/src/wallet_api.rs]: fixed

File: forest/cli/src/cli/wallet_cmd.rs (hunk 1 above)
Comment (lemmih) [reply forest/cli/src/cli/wallet_cmd.rs:225]: I'm thinking:
```rust
if default.as_ref() == Some(&addr) { "X" } else { "" };
```

Comment (creativcoder) [reply forest/cli/src/cli/wallet_cmd.rs]: that's more lean, thx.

File: forest/cli/src/cli/wallet_cmd.rs [hunk 3]
```diff
@@ -155,7 +155,8 @@ impl WalletCommands {
             Self::Default => {
                 let response = wallet_default_address((), &config.client.rpc_token)
                     .await
-                    .map_err(handle_rpc_err)?;
+                    .map_err(handle_rpc_err)?
+                    .unwrap_or("No default wallet address set".to_string());
```
[the remark this answers is not in the document]
Comment (creativcoder) [reply forest/cli/src/cli/wallet_cmd.rs]: using `unwrap_or_else`

**Review by lemmih — APPROVED** [review]

File: forest/cli/src/cli/send_cmd.rs [hunk 4]
```diff
@@ -38,7 +38,10 @@ impl SendCommand {
             Address::from_str(
                 &wallet_default_address((), &config.client.rpc_token)
                     .await
-                    .map_err(handle_rpc_err)?,
+                    .map_err(handle_rpc_err)?
+                    .ok_or(anyhow::anyhow!(
```
Comment (lemmih) [inline forest/cli/src/cli/send_cmd.rs:42]:
```suggestion
                    .ok_or_else(|| anyhow::anyhow!(
```

**Pull request merged**